### PR TITLE
Convert to new plugin API

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

File: ios/Classes/GoogleSignInPlugin.m [hunk 1]
```diff
@@ -3,36 +3,45 @@
...
-- (instancetype)initWithController:(FlutterViewController *)controller {
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FlutterMethodChannel *channel =
+  [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/google_sign_in"
+                              binaryMessenger:[registrar messenger]];
+  // TODO(goderbauer): cast is workaround for https://github.com/flutter/flutter/issues/9961.
+  UIViewController *viewController = (UIViewController *)registrar.messenger;
+  GoogleSignInPlugin *instance = [[GoogleSignInPlugin alloc] initWithViewController: viewController];
+  [registrar addApplicationDelegate:instance];
+  [registrar publish: instance];
```
Comment (mravn-google) [inline ios/Classes/GoogleSignInPlugin.m]: Remove space after :

**Comment by eseidelGoogle** [comment]
Given the state of the engine yesterday, I suspect it's too late/risky, but I defer to @Hixie who managed the alpha release for I/O.

Also, did the new plugin changes get rolled in before or after alpha was cut?  Does alpha still work with these updated plugins?

**Comment by Hixie** [comment]
I'm all about taking risks, so I'm happy to roll alpha again if we think it's important enough. Engine was rolled about 4 hours ago, do you want another roll beyond that?

Can you elaborate on how much it improves the codelab? What's the actual difference? Do you think it's worth the risk? The changes since we cut alpha include a skia roll, but very little else that's particularly risky so far.

**Comment by collinjackson** [comment]
The difference is that it lets users avoid writing any Objective-C in the codelab.

**Comment by collinjackson** [comment]
Looks like we need to roll engine again to get https://github.com/flutter/engine/commit/006ae2fef4906b31e2ed1a81e8010628ed2c0c74

Personally, I think it's worth the risk, but I don't feel strongly about it.

**Comment by goderbauer** [comment]
It works with @mravn-google's fix. I removed the workaround since that is now gonna be part of the alpha branch. Thanks for the quick turn-around time!

**Pull request merged**